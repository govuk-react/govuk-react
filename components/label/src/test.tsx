import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import Label from '.';

describe('Label', () => {
  const example = 'example';
  const wrapper = <Label>{example}</Label>;
  const withError = <Label error={!!example}>{example}</Label>;

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });

  it('matches withError snapshot', () => {
    expect(mount(withError)).toMatchSnapshot('with error mount');
  });
});
