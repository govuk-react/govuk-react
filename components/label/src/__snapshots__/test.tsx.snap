// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label matches withError snapshot: with error mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  border-left: 4px solid #d4351c;
  margin-right: 15px;
  padding-left: 10px;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<Label
  error={true}
>
  <src__StyledLabel
    error={true}
  >
    <label
      className="c0"
    >
      example
    </label>
  </src__StyledLabel>
</Label>
`;

exports[`Label matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

<Label
  error={false}
>
  <src__StyledLabel
    error={false}
  >
    <label
      className="c0"
    >
      example
    </label>
  </src__StyledLabel>
</Label>
`;
